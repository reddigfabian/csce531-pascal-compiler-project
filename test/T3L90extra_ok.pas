(* Test file for level 90 of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L90extra_ok;

var
   a,b : Integer;
   c   : Char;


procedure print_one; External;
procedure print_two; External;
procedure print_three; External;
procedure print_dunno; External;
procedure print_big; External;
procedure print_just; External;
procedure print_isnt; External;
procedure print_hello; External;
procedure print_oops; External;
procedure print_712; External;
procedure print_is_the; External;
procedure print_grooviest; External;
procedure print_newline; External;


begin

   a := 3;

   case a of
     5..10: print_oops;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }

   a := 4;

   case a of
     5..10: print_big;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }
   
   a := 7;

   case a of
     5..10: print_big;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }
   
   print_newline

end.
