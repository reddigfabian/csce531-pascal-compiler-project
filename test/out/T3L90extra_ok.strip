b_global_decl (B, alignment = 4, size = 4)
b_global_decl (A, alignment = 4, size = 4)
b_global_decl (C, alignment = 1, size = 1)
b_func_prologue (main)
b_push_ext_addr (A)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  < ,signed long int, 5, .LC3, no pop on jump )
b_jump ( destination = .LC3 )
b_dispatch (  <= ,signed long int, 10, .LC2, pop on jump )
b_jump ( destination = .LC2 )
b_jump ( destination = .LC1 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC0 )
b_dispatch (  < ,signed long int, 1, .LC8, no pop on jump )
b_jump ( destination = .LC8 )
b_dispatch (  <= ,signed long int, 3, .LC7, pop on jump )
b_jump ( destination = .LC7 )
b_jump ( destination = .LC6 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC0 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (4)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  < ,signed long int, 5, .LC14, no pop on jump )
b_jump ( destination = .LC14 )
b_dispatch (  <= ,signed long int, 10, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_jump ( destination = .LC12 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_big, void)
b_jump ( destination = .LC11 )
b_dispatch (  < ,signed long int, 1, .LC19, no pop on jump )
b_jump ( destination = .LC19 )
b_dispatch (  <= ,signed long int, 3, .LC18, pop on jump )
b_jump ( destination = .LC18 )
b_jump ( destination = .LC17 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC11 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_push_ext_addr (A)
b_push_const_int (7)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_dispatch (  < ,signed long int, 5, .LC25, no pop on jump )
b_jump ( destination = .LC25 )
b_dispatch (  <= ,signed long int, 10, .LC24, pop on jump )
b_jump ( destination = .LC24 )
b_jump ( destination = .LC23 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_big, void)
b_jump ( destination = .LC22 )
b_dispatch (  < ,signed long int, 1, .LC30, no pop on jump )
b_jump ( destination = .LC30 )
b_dispatch (  <= ,signed long int, 3, .LC29, pop on jump )
b_jump ( destination = .LC29 )
b_jump ( destination = .LC28 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_one, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_two, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_three, void)
b_jump ( destination = .LC22 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dunno, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_func_epilogue (main)
