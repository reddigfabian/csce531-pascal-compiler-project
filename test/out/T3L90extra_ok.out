a is one
a is two
a is three
dunno what a is
a is big
...
